src/iam/iam-base.ts

```diff
@@ -199,98 +195,127 @@ export class IAMBase {
     this.setupENS();
   }
 
-  private async setupProvider({
+  private async initSigner({
     useMetamask,
     initializeMetamask
   }: {
     useMetamask: boolean;
     initializeMetamask?: boolean;
   }) {
-    this._provider = new providers.JsonRpcProvider(this._connectionOptions.rpcUrl);
+    if (!this._provider) {
+      throw new Error("Initialization of singer failed due to no provider");
+    }
     if (this._connectionOptions.privateKey) {
       this._signer = new Wallet(this._connectionOptions.privateKey, this._provider);
-      const publicKey = await this.getPublicKey();
-      this._didSigner = new IdentityOwner(
-        this._signer,
-        this._provider,
-        publicKey,
-        this._connectionOptions.privateKey
-      );
       return;
     }
-    if (this._runningInBrowser) {
-      const metamaskProvider = (await detectMetamask({
-        mustBeMetaMask: true
-      })) as any;
-      if (metamaskProvider && useMetamask) {
-        if (initializeMetamask) {
-          await metamaskProvider.request({
+    if (!this._runningInBrowser) {
+      throw new Error("Initialization of signer not possible in nodejs env without private key ");
```

Comment:
I would add a new entry to `ERROR_MESSAGES` to use here.

Reply:
👍